catalog/app/utils/Data.js

```diff
@@ -47,7 +49,13 @@ export function useData(request, params, { noAutoFetch = false } = {}) {
     else fetch()
   })
 
-  const result = useMemoEq(state, mapResult)
+  usePreviousSync(params, (prev) => {
+    if (!R.equals(params, prev)) {
+      stateResetingIsInProgress = true
+    }
+  })
+
+  const result = useMemoEq(stateResetingIsInProgress ? initial : state, mapResult)
```

Comment:
this is not quite correct, bc if auto-fetch is enabled we should transition directly to `Pending` (skipping `Init`)